provider/storage_benchmark.py

```diff
@@ -0,0 +1,660 @@
+    logging.debug(o)
+    if not s:
+        return o.splitlines()[0]
+    return None
+
+
+def find_executable_exe(session, find_dir, name, timeout=300):
+    """
+    Find the path of the given executable file in windows.
+
+    :param session: session connected guest
+    :param find_dir: directory to be found
+    :type find_dir: str
+    :param name: name of executable file
+    :type name: str
+    :param timeout: timeout for finding
+    :type timeout: float
+    :return: full path of tool
+    :rtype: str
+    """
+    s, o = session.cmd_status_output(
+        r'CD %s && DIR /S /B %s.exe' % (find_dir, name), timeout=timeout)
+    logging.debug(o)
+    if not s:
+        return '"{}"'.format(o.splitlines()[0])
+    return None
+
+
+class Benchmark(object):
+    """
+    Create a Benchmark class which provider common interface(method) for
+    using Benchmark tool to run test.
+
+    """
+    def __init__(self, test, params, session, name):
+        """
+        :param test: kvm test object
+        :param params: dictionary with the test parameters
+        :param session: session connected guest
+        :param name: name of benchmark
+        :type name: str
+        """
+        self.test = test
+        self.session = session
+        self.params = params
+        self.name = name
+        self.os_type = params['os_type']
+        self.setup_files = []
+        self._do_clean = False
+        self.rm_timeout = float(params.get('rm_timeout', '300'))
+        self.ret_path = ''
+
+    def setup(self, func):
+        """
+        Set up the given benchmark tools.
+
+        :param func: the setup function such as installation
+        :type func: callable object
+        """
+        logging.info("Starting to setup %s." % self.name)
+        try:
+            ret = func()
+            self._do_clean = True
+            return ret
+        except Exception as e:
+            remove_files_force(self.session, self.os_type,
+                               self.setup_files, self.rm_timeout)
+            self.test.error(str(e))
+
+    def run(self, cmd, timeout=600, record=True, debug=True):
+        """
+        Execute the given command.
+
+        :param cmd: executed command
+        :param timeout: timeout for executing command
+        :type timeout: float
+        :param record: if True, write the output to a log file
+        :type record: bool
+        :param debug: if True, print the output of executed command
+        :type debug: bool
+        :return: output of running command
+        :rtype: str
+        """
+        random_string = utils_misc.generate_random_string(8)
+        stamp = time.strftime('%Y-%m-%d-%H-%M-%S') + random_string
+
+        status, output = self.session.cmd_status_output(cmd, timeout=timeout)
+        if debug:
+            logging.debug(output)
+        if record:
+            ret_name = ''.join(('-'.join((self.name, 'test', stamp)), '.log'))
```

Comment:
`'%s-test-%s.log' % (self.name, stamp)` 
is easier to read, and you probably dont need the `random_string`.